reactor-core/src/main/java/reactor/core/publisher/FluxSwitchOnFirst.java

```diff
@@ -618,7 +462,9 @@ boolean drainRegular() {
             T f = first;
             int m = 1;
             boolean sent = false;
-            Fuseable.ConditionalSubscriber<? super T> a = inner;
+            @SuppressWarnings("unchecked")
+            Fuseable.ConditionalSubscriber<? super T> a =
+                    (Fuseable.ConditionalSubscriber<? super T>) inner;
```

Comment:
can't comment on the proper line, but ~7 lines below, `if (cancelled)` before the `tryOnNext`, shouldn't it `return false`?

Reply:
fixed